File: 6_Bitmanip/C++/lc1863.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        vector<int> res;
        dfs(0, nums, {}, res);
        int sum = 0;
        for (int num : res) {
            sum += num;
        }
        return sum;
    }

private:
    void dfs(int start, vector<int>& arr, vector<int> current, vector<int>& res) {
        int x = 0;
        for (int num : current) {
            x ^= num;
        }
        res.push_back(x);

        for (int i = start; i < arr.size(); i++) {
            current.push_back(arr[i]);
            dfs(i + 1, arr, current, res);
            current.pop_back();
        }
    }
};

File: 6_Bitmanip/Py/lc1863.py
from typing import List


class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        res = []

        def dfs(start, arr, l):
            x = 0
            for i in l:
                x ^= i
            res.append(x)

            for i in range(start, len(arr)):
                l.append(arr[i])
                dfs(i+1, arr, l)
                l.pop()

        dfs(0, nums, [])
        return sum(res)
    
    